tree/ntuple/v7/inc/ROOT/RColumn.hxx

```diff
@@ -176,51 +176,33 @@ public:
 
    void Read(const RClusterIndex &clusterIndex, void *to)
    {
-      if (!fReadPage.Contains(clusterIndex)) {
-         MapPage(clusterIndex);
-      }
-      const auto elemSize = fElement->GetSize();
-      void *from = static_cast<unsigned char *>(fReadPage.GetBuffer()) +
-                   (clusterIndex.GetIndex() - fReadPage.GetClusterRangeFirst()) * elemSize;
-      std::memcpy(to, from, elemSize);
+      Read(GetGlobalIndex(clusterIndex), to);
    }
 
-   void ReadV(const NTupleSize_t globalIndex, const ClusterSize_t::ValueType count, void *to)
+   void ReadV(NTupleSize_t globalIndex, ClusterSize_t::ValueType count, void *to)
    {
-      if (!fReadPage.Contains(globalIndex)) {
-         MapPage(globalIndex);
-      }
-      NTupleSize_t idxInPage = globalIndex - fReadPage.GetGlobalRangeFirst();
+      while (count > 0) {
+         if (!fReadPage.Contains(globalIndex)) {
+            MapPage(globalIndex);
+         }
+         NTupleSize_t idxInPage = globalIndex - fReadPage.GetGlobalRangeFirst();
 
-      const auto elemSize = fElement->GetSize();
-      const void *from = static_cast<unsigned char *>(fReadPage.GetBuffer()) + idxInPage * elemSize;
-      if (globalIndex + count <= fReadPage.GetGlobalRangeLast() + 1) {
-         std::memcpy(to, from, elemSize * count);
-      } else {
+         const auto elemSize = fElement->GetSize();
```

Comment:
Should we move it outside the loop?

Reply:
Yes, we can. In principle also the `idxInPage` which we know is 0 for all but the first iteration of the loop, but I'm not sure how much we should micro-optimize here...